tempesta_fw/connection.c

```diff
@@ -88,10 +88,15 @@ tfw_connection_release(TfwConn *conn)
 }
 
 /*
- * Code architecture decisions ensure that conn->sk remains valid
- * for the life of @conn instance. The socket itself may have been
- * closed, but not deleted. ss_send() makes sure that data is sent
- * only on an active socket.
+ * Send @msg through connection @conn. Code architecture decisions
+ * ensure that conn->sk remains valid for the life of @conn instance.
+ * The socket itself may have been closed, but not deleted. ss_send()
+ * makes sure that data is sent only on an active socket.
+ * Return value:
+ *   0		- @msg had been sent successfully;
+ *   -EBADF	- connection is broken;
+ *   -EBUSY	- transmission work queue is full;
+ *   -ENOMEM	- out-of-memory error occured.
```

Comment:
In case of `-EBUSY` error we have the problem on current work queue and can try with another one. But in case `-ENOMEM` we have the problem with memory lacking, and with high probability we will not succeed on other work queues, so it seems that would be better to drop requests (with error responses or without them) - to avoid additional memory overload and, probably, to allow the system to recover free memory.